operator/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+    pub fn new() -> KafkaStrategy {
+        KafkaStrategy {}
+    }
+}
+
+#[async_trait]
+impl ControllerStrategy for KafkaStrategy {
+    type Item = KafkaCluster;
+    type State = KafkaState;
+    type Error = error::Error;
+
+    fn finalizer_name(&self) -> String {
+        FINALIZER_NAME.to_string()
+    }
+
+    async fn init_reconcile_state(
+        &self,
+        context: ReconciliationContext<Self::Item>,
+    ) -> Result<Self::State, Self::Error> {
+        let existing_pods = context.list_pods().await?;
+        trace!("Found [{}] pods", existing_pods.len());
+
+        Ok(KafkaState {
+            kafka_spec: context.resource.spec.clone(),
+            context,
+            zk_spec: None,
+            existing_pods,
+        })
+    }
+}
+
+pub async fn create_controller(client: Client) {
+    let kafka_api: Api<KafkaCluster> = client.get_all_api();
+    let pods_api: Api<Pod> = client.get_all_api();
+    let config_maps_api: Api<ConfigMap> = client.get_all_api();
+
+    let controller = Controller::new(kafka_api)
+        .owns(pods_api, ListParams::default())
+        .owns(config_maps_api, ListParams::default());
+
+    let strategy = KafkaStrategy::new();
+
+    controller.run(client, strategy).await;
+}
+
+fn build_pod(
+    resource: &KafkaCluster,
+    broker: &KafkaBroker,
+    labels: &BTreeMap<String, String>,
+    pod_name: &str,
+    cm_name: &str,
+) -> Result<Pod, Error> {
+    let pod = Pod {
+        // Metadata
+        metadata: metadata::build_metadata(pod_name.to_string(), Some(labels.clone()), resource)?,
+        // Spec
+        spec: Some(PodSpec {
+            node_name: Some(broker.node_name.clone()),
+            tolerations: Some(stackable_operator::create_tolerations()),
+            containers: vec![Container {
+                image: Some(format!(
+                    "stackable/kafka:{}",
+                    serde_json::json!(resource.spec.version).as_str().unwrap()
+                )),
+                name: "kafka".to_string(),
+                command: Some(vec![
+                    "kafka_2.12-2.6.0/bin/kafka-server-start.sh".to_string(), // TODO: Don't hardcode this
+                    "{{ configroot }}/config/server.properties".to_string(),
+                ]),
+                volume_mounts: Some(vec![VolumeMount {
+                    mount_path: "config".to_string(),
+                    name: "config-volume".to_string(),
+                    ..VolumeMount::default()
+                }]),
+                ..Container::default()
+            }],
+            volumes: Some(vec![Volume {
+                name: "config-volume".to_string(),
+                config_map: Some(ConfigMapVolumeSource {
+                    name: Some(format!("{}-config", cm_name)),
+                    ..ConfigMapVolumeSource::default()
+                }),
+                ..Volume::default()
+            }]),
+            affinity: Some(Affinity {
+                pod_anti_affinity: Some(PodAntiAffinity {
```

Comment:
This is currently not needed since you schedule the pod yourself, correct?
I'm fine with it staying in, but if that is the case maybe add a very brief comment.